### PR TITLE
Refactor chemistry application instances

### PR DESCRIPTION
**Pull request by max-radin**
## Description

The chemistry application instances had some code duplication and inconsistent patterns. This PR addresses this by introducing a `ChemistryApplicationInstance` dataclass based on discussion with @aakunitsa and @mkodrycka. The dataclass encapsulates the information that defines an application instance, such as geometry, charge, and active space.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: src/benchq/problem_ingestion/molecule_instance_generation.py [hunk 1]
```diff
@@ -9,15 +12,65 @@
...
-    # mean_field_object = scf.UHF(molecule)
-    # mean_field_object = scf.ROHF(molecule)
-    mean_field_object = scf.ROHF(molecule)
+from openfermionpyscf import PyscfMolecularData, run_pyscf
+from pyscf import scf
+
+
+@dataclass
+class ChemistryApplicationInstance:
+    """Class for representing chemistry application instances."""
```
Comment (mstechly) [inline src/benchq/problem_ingestion/molecule_instance_generation.py:21]: Could you add more to the docstrings?
For me it's unclear what all this means. It doesn't need to be a lecture in quantum chemistry, but perhaps even a reference to a specific place in pyscf docs would be helpful :)

File: tests/benchq/problem_ingestion/test_molecule_instance_generation.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import pytest
+
+from benchq.problem_ingestion import (
+    WATER_MOLECULE,
+    ChemistryApplicationInstance,
+    generate_hydrogen_chain_instance,
+)
+
+
+@pytest.mark.parametrize(
```
Comment (mstechly) [inline tests/benchq/problem_ingestion/test_molecule_instance_generation.py:10]: Tests look a bit sparse – I'm not sure what exactly should be included here, but it looks to me that some of the input arguments to the `ChemistryApplicationInstance` are not covered by tests at all, e.g.: AVAS ones.

File: src/benchq/problem_ingestion/molecule_instance_generation.py (hunk 1 above)
Comment (max-radin) [reply src/benchq/problem_ingestion/molecule_instance_generation.py]: Fixed in 48aaa1100418da063558325d823854abf950498e

Comment (max-radin) [reply src/benchq/problem_ingestion/molecule_instance_generation.py]: Fixed in 48aaa1100418da063558325d823854abf950498e and 9eefbbc5907d9ff2f3b5e3972f25a940600a6609

File: tests/benchq/problem_ingestion/test_molecule_instance_generation.py (hunk 2 above)
Comment (max-radin) [reply tests/benchq/problem_ingestion/test_molecule_instance_generation.py:9]: The AVAS arguments were actually covered by the water molecule test. But I refactored it to have the test explicitly set the AVAS options in 9f2cc32c779118dbccfe6fa4e6bf20a9082a12a0 and also added tests for cases that are invalid.

**Review by mstechly — APPROVED** [review]

**Pull request merged**